src/devices/cpu/upd177x/upd177x.cpp

```diff
@@ -0,0 +1,797 @@
+// license:BSD-3-Clause
+// copyright-holders:Wilbert Pol
+/*
+
+TODO:
+- Only upd1771c supported
+- Noise, time, and external interrupts not implemented
+- CH1/CH2 not implemented
+- Not all instructions implemented
+- Are PNC1 and PNC2 updated every machine cycle or only when the noise counter overflows?
```

Comment:
PNC1/2 are updated when a specific bit of the noise counter (selected by MD_NSF2/3) is cleared (negedge). That specific bit is bit 8 by default, so the update event precisely coincides with the noise counter overflowing. (The PNC1/2 update event is also the noise interrupt trigger.)

Reply:
That is how I have it currently implemented.
In the description of the MIX command there is some text mentioning that the PNC2 output is updated every machine cycle. That is where my confusion came from. 
Thanks for confirming that following the noise counter is correct.